os/drivers/kernel_test/test_net_pbuf.c

```diff
@@ -0,0 +1,67 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <debug.h>
+
+#include <tinyara/kernel_test_drv.h>
+#include <tinyara/sched.h>
+#include <lwip/pbuf.h>
+
+typedef struct pbuf_test_args pbuf_test_args_t;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int test_pbuf(unsigned long arg)
+{
+	int ret = 0;
+	struct pbuf* p = NULL;
+	pbuf_test_args_t *args;
+	args = (void *)kmm_malloc(sizeof(pbuf_test_args_t));
+	memcpy(args, arg, sizeof(pbuf_test_args_t));
+
+	p = pbuf_alloc(args->layer, args->len, args->type);
+	if (p) {
+		ret = 1;
+	}
+
+	pbuf_free(p);
+	return ret;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int test_net_pbuf(int cmd, unsigned long arg)
+{
+	int ret = -1;
+
+	switch(cmd) {
```

Comment:
Do you need to use a switch-case construct if there is only one "cmd" to be checked?
Can simply use if-else unless more commands are to be added in the near future for expansion.

```
if (cmd == TESTIOC_NET_PBUF) {
        return test_pbuf(arg);
} else {
        return -1;
}
```

Reply:
in future, we may add more commands this is why we used switch.